### DISCUSSION TITLE
Welcome to ActivityManager Discussions!

### DISCUSSION THREAD

**Discussion by sdex** [sdex/ActivityManager#29] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by drogga** [comment]
https://galaxystore.samsung.com/detail/com.activitymanager > 4.1.0 (2021.08.20) > too old, please update it if you can and perhaps include the link as available option in the ReadMe. Thank You.

**Reply by sdex** [reply]
Uploaded. 
I completely forgot how bad the galaxy store seller portal is.

**Reply by drogga** [reply]
I guess you won't be linking it to the ReadMe here..., I can create a PR for that, but I'm not sure I know how to use the Galaxy Store button image and where to find it.

**Reply by sdex** [reply]
The readme used to have links to galaxy store and appgallery. 
I'm not going to maintain them and update the app regularly, so do not want to add them back.

**Reply by sdex** [reply]
https://github.com/sdex/ActivityManager/tree/master/docs/graphics/logos

**Comment by drogga** [comment]
@sdex

I sometimes miss the separation lines and the way the App Name and the Package Name were in different colors (White vs. Gery) in the older pre-Material v3 vers., please consider making the P.N. & Version texts in different color again. Thanks.

I was surprised that it can't launch its own unexported activities (Settings or About for example), "Activity Launcher" for example can launch his, also without Root.
There's a typo btw "Root is not available or the permission is not granter", should be "granted" [I've fixed it in a PR].
I wonder if it will make sense to have the ability/option to list the unexported last (on the bottom)...

**Reply by sdex** [reply]
I felt relieved when I finally removed the dividers. Also, I declined to use custom colors for components and turned to material design (with future dynamic colors support).

The app treats all non-exported activities the same. There wasn't a reason to make this exclusion.

**Reply by drogga** [reply]
What do you think about making the P.N. & Ver. italic ?

**Reply by sdex** [reply]
Why? I used it to show the system/disabled and it was ugly.

**Reply by drogga** [reply]
Because I think it looked and will look better if the second line is more subtle, because in my opinion it's currently a bit confusing, so I would like to better differentiate them with the app name and IDT an even smaller font will help _(I might be wrong tho and making it look like the version line would help)_, but if you think it's pointless and don't like how it looks with a bit greyed out color for the p.n. only, then NVM, forget about it, I will drop the idea (which I can't implement on my own).

BTW, you've previously wrote: "_The app treats all non-exported activities the same. There wasn't a reason to make this exclusion._", but I'm not sure is it an answer to the fact that it can't launch it's own unexported activities _(I don't see a reason why not, given that it doesn't actually need higher privileges to do this, like root or how some are using Shizuku, but it actually doesn't help launching such, only device admin/owner, like a Knox (MDM) app on Samsung's like how "Alliance Shield X" used to be able to launch any no matter what, before they ruined it, since the Knox license was revoked or deprecated)_ or was an answer to me wondering if it will be better if those are listed on the bottom of the activities menu/screen ?

The "Support Development [Donate]" banner is in white color even with the dark theme, can you pls. make it more subtle and not so annoying or place a button to close it forever ?, because it looks like it decides to appear all of a sudden and after like ~10 launches (and clearing the app from the recents) it hides itself, until I update and it appears again, so I would really appreciate if there's a way to manually close/dismiss/hide it **forever** with a gesture or a button on it.

**Reply by sdex** [reply]
> it can't launch it's own unexported activities

It can, just doesn't want to. 

> a way to manually close/dismiss/hide it with a gesture 

You can swipe it to the left or right to dismiss.

**Reply by drogga** [reply]
"It can, just doesn't want to." > Obv., you must have a good reason to purposely restrict it like that, despite the nature of the app, I might not exactly get why and think it's unnecessary, but OK, NVM.

Yup, it appears that I can swipe to close it, but it's on the persistent / sticky side, because it appears again when I back out and launch again, unless I swipe away the app from the recents, then it stays hidden for some amount of launches, plus the rest of the points are still valid about it, that it's in white, which doesn't look good with the dark theme, its design doesn't merge well with the rest of the UI and it looks a bit dated and on the intrusive side (sorry about the constructive criticism).

**Reply by sdex** [reply]
I will probably remove it, no one donates anyway.

**Reply by drogga** [reply]
I would, if I wasn't poor AF (not that I expect anyone to believe, but it's true !)

**Reply by drogga** [reply]
I see that you removed the donate snackbar and then changed your mind and reverted the change and restored it...

**Reply by sdex** [reply]
Yeah, I want it to stick around and annoy people a bit.

**Reply by drogga** [reply]
Consider me irritated/annoyed :D

Please consider changing how the context menu shows up, because having to swipe it up to show it is not an option with a remote and I either have to do that with a mouse or touch interaction, which on a device connected to a TV is not always an option and quite honestly it's pointless/unnecessary and annoying being that way in apps even on a device with a touchscreen, IDK how I'm even supposed to swipe away the donate snackbar with a remote when it shows up (I uninstalled it from the TV because of this), the other UI elements/things were not a problem to navigate tru with the remote's D-Pad.

BTW, it didn't build with just `./gradlew assemble` last week when I tried on Linux, Installed Gradle 8.5-1 with dependencies and OpenJDK 21, it DLed 8.5 when I ran it, but it still failed to build, IMO you should try with BigLinux Live Boot for example (just like I did) and do the necessary changes to the code/ReadMe, so the instructions to be correct and to actually build/compile with that command (I'm not a Linux user and if additional setup and config. is necessary, then it should be mentioned, because else IDK what I'm doing and how I'm supposed to built it).

**Reply by sdex** [reply]
`./gradlew assemble` works for me.

**Reply by drogga** [reply]
...and you did just installed gradle (clean slate) on linux ?
Should I try again and record a video this time ?

**Reply by sdex** [reply]
Besides gradle, you need Android SDK. 
I didn't build the project without Android Studio though.

**Reply by drogga** [reply]
That's the point, to build it without Android Studio or any other IDE (compiler with a GUI, not just a command line based one), else I don't see any point in this command being in the ReadMe, because it's misleading, making you think that with only gradle and JDK it can be build, or perhaps I'm the only one that got confused and don't have experience or knowledge of how the things work on the development side, since I'm not a coder.

**Reply by sdex** [reply]
Yep, it will work without any IDE. Just with Gradle, JDK, and Android SDK.

**Reply by drogga** [reply]
@sdex

This one can now with the latest ver. actually launch unexpected activities for me with Shizuku (it's Paid on PS): https://github.com/zacharee/RootActivityLauncher, please add Shizuku, Dhizuku or any other non-root method to launch such (when possible) in your app as well, previously I mentioned that implementing those is basically pointless, but it now works (IDK why it didn't before, probably a bad implementation...), so I take my words back about using Shizuku in particular for this.

When I mentioned that it's not practical and really annoying having to drag up to show the sheet from the bottom, I meant in Landscape orientation, I should've explicitly mentioned that - sorry.

For me on S20 FE, the Sort filter's last buttons for Disabled applications > Show & Label ones are partially hidden when I drag that sheet up to reveal it in landscape.

The apps context sheet can and should IMO also have a "Create Shortcut" button (of the main activity), even tho you can just drag an app's icon from the drawer to whatever you want on the home screen, but given that there's now a button under Show non-exported for this, I think that it makes sense to be in the main menu as well.

The new "Create shortcut" dialog style doesn't exactly follow the color palette now, here on A.11 (it is what it is, no way to change it on a system level - globally, like in A.12+ or in the app), it was better before, just not with that much rounded corners, which was fine for me. Now it's the same as the top 3-dots context menu (which is with some sort dark purple color or something like that - I don't think it's necessary to mention it's hex code here of what I see...).

It's better if the "Open in GooglePlay" button is called and renamed to "Open in Play Store", since that's the actual app name and most known as.

I think you should really think about making all the sheet dialogs a pop-up centered ones (just like the "Create shortcut" one), it will be a lot better like this, not only on Phones & Tablets.

Sometimes when I go to the app's activities list, 1 icon of this app's all activities shows up crippled and I have to remove it from the recents and launch it again in order to show all properly, probably also happens with other apps, where either they are with large icons (.png or vector), or it's something else, but most times I noticed it is with Activity Manager itself. To reproduce it, try going back and forth to the app's activities list menu.

Please make the NavBar transparent or translucent so it doesn't cover part of the items, for example: check out "SetEdit" by Muntashir Akon, here on GH.

Commit: "add an option to disable donate reminder" > I don't see it anywhere, nor that snackbar's style changed, despite the commits saying that it did.

**Reply by drogga** [reply]
@sdex When an app which activity has been just launched crashes (happens often with unexported/private and root), Activity Manager goes back to the all apps list, instead of staying exactly where it was in the activities list of the particular app, your app is not the one that crashes in those cases, so I don't understand why it doesn't retain the view and stay where it was before the launching of the activity that caused the target app to crash.

**Reply by sdex** [reply]
It's quite strange and should not happen. Can you give any reproducible example?

**Reply by drogga** [reply]
Not anything in particular, sorry, I was playing with A.8.1 x86 (it's rooted) live boot on a PC the other day when I reported this and don't remember which activities of which app I was playing with/launched, but it should be easily reproducible.

**Reply by drogga** [reply]
5.4.12:
> move app preferences to the dialog
> fix preferences dialog expand/collapse states for landscape orientation

**Please** do the same for the context (3-dots / long press) menus of apps / activities.
Consider a Black (Amoled) Theme in future, which will also turn the top blue to completely black.
Thank You in advance.

**Reply by sdex** [reply]
> Please do the same for the context (3-dots / long press) menus of apps / activities.

OK, it will be in the next update. 

> Consider a Black (Amoled) Theme in future

No themes/colors updates are planned at all, I'm tired of it.
